pilot/pkg/serviceregistry/kube/controller.go

```diff
@@ -93,6 +105,17 @@ type Controller struct {
 	// Env is set by server to point to the environment, to allow the controller to
 	// use env data and push status. It may be null in tests.
 	Env *model.Environment
+
+	// ClusterID identifies the remote cluster in a multicluster env.
+	ClusterID string
```

Comment:
VNID..

Reply:
No networks here.

Reply:
Well, you have to pick a consistent terminology, since you are cleaning up or establishing a new model. ClusterID in one place, shard in another place, and then another VNID in future -- how do they all fit together? Even within this PR, clusterID and endpoint shard names - are tehy different or same? if they are different, how so?

Reply:
I re-used ClusterID from aggregate.Registry - that's how the different registries are configured. Not happy with the old name, I just reused it. It's really 'RegistryID' - in particular since Consul registry can map multiple 'clusters'. Happy to change in master. 

VNID can be debated in the split horizon PR, and will be master only.  I tried to not mention or use anything network-related in this PR, to not create problems for that PR.

Shard is a general concept - I will expects other parts of the code to be sharded (out clusters for example - when we optimize CDS). There is no need to recompute the whole list when only one rule changes.